### PR TITLE
Only compute step loss on non-same-day reviews

### PR DESCRIPTION
**Pull request by joshdavham**
Previously, the optimizer would only compute step loss on review-state cards. For example:

```python
if card.state == State.Review:
    step_loss = loss_fn(y_pred_retrievability, y_retrievability)
```

However, it makes more sense to more generally compute step loss on non-same-day (long term) reviews rather than depending on the specific learning state of the card. Technically, in the past, a user could review a review-state card multiple times in one day and this fix simply accounts for that case.

Now, step loss is computed on all non-same-day reviews, not just review-state cards.
```python
if card.last_review and (review_datetime - card.last_review).days > 0:
    step_loss = loss_fn(y_pred_retrievability, y_retrievability)
```

I also adjusted one of the unit tests as the optimal computed parameters are now slightly different.

Let me know if you have any questions 👍

### CODE REVIEW

File: fsrs/optimizer.py [hunk 1]
```diff
@@ -170,7 +170,7 @@ def _compute_batch_loss(self, parameters: list[float]) -> float:
                         y_retrievability, dtype=torch.float64
                     )
 
-                    if card.state == State.Review:
+                    if card.last_review and (x_date - card.last_review).days > 0:
```
Comment (L-M-Sherlock) [inline fsrs/optimizer.py:173]: Does it consider rollover? I mean something like `next_day_start_at` of Anki.

Comment (joshdavham) [reply fsrs/optimizer.py:173]: > Does it consider rollover?

Currently no, but that could potentially be a cool feature to add in the future. Maybe in a future release, we could add a `day_starts_at` attribute or something to the `Scheduler`.

However, retrievability is currently calculated like this https://github.com/open-spaced-repetition/py-fsrs/blob/816c9d628b04ea3de9601663b8c481c4c854d0e8/fsrs/fsrs.py#L232
meaning that it's simply equal to 1 when elapsed days are 0 (same-day review).

This pr is just trying to make the optimizer consistent with how retrievability is computed when `elapsed_days=0`.

**Review by L-M-Sherlock — APPROVED** [review]
LGTM

**Comment by L-M-Sherlock** [comment]
For the calculation of elapsed days, you can refer to:

- https://github.com/open-spaced-repetition/ts-fsrs/pull/144
